implementations/rust/ockam/ockam_command/src/error.rs

```diff
@@ -1,59 +1,157 @@
+        url("https://docs.ockam.io/errors/OCK503")
+    )]
+    #[error("{resource} {resource_name} is unavailable")]
+    Unavailable {
+        resource: String,
+        resource_name: String,
+    },
+    // ==== End 5xx Errors ====
 }
 
 impl Error {
-    pub fn new(code: ExitCode, err: anyhow::Error) -> Self {
+    pub fn new(code: ExitCode, err: miette::ErrReport) -> Self {
         assert_ne!(code, 0, "Error's exit code can't be OK");
-        Self {
-            code,
-            description: err.to_string(),
-            cause: err.source().map(|s| s.to_string()),
+        Error::InteralError {
+            error_message: err.to_string(),
+            exit_code: code,
         }
     }
 
-    pub fn code(&self) -> ExitCode {
-        self.code
+    pub fn new_software_error(human_err: &str, inner_err_msg: &str) -> Self {
+        let msg = format!("{}\n{}", human_err, fmt_log!("{}", inner_err_msg));
+        Self::new(exitcode::SOFTWARE, miette!("{}", msg))
     }
-}
 
-impl Display for Error {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "{}", Version::short())?;
-        let description = &self.description;
-        if let Some(cause) = &self.cause {
-            writeln!(f, "{}", fmt_err!("{description}. Caused by: {cause}"))?;
-        } else {
-            writeln!(f, "{}", fmt_err!("{description}"))?;
+    pub fn code(&self) -> ExitCode {
+        match self {
+            Error::NotFound { .. } => exitcode::SOFTWARE,
+            Error::Unauthorized { .. } => exitcode::NOPERM,
+            Error::Conflict { .. } => exitcode::SOFTWARE,
+            Error::InteralError { exit_code, .. } => *exit_code,
+            Error::Unavailable { .. } => exitcode::UNAVAILABLE,
         }
-        Ok(())
     }
 }
 
-impl std::error::Error for Error {}
-
 impl From<anyhow::Error> for Error {
     fn from(e: anyhow::Error) -> Self {
-        Error::new(exitcode::SOFTWARE, e)
+        Error::new(exitcode::SOFTWARE, miette!(e.to_string()))
+    }
+}
+
+pub struct ErrorReportHandler;
+impl ErrorReportHandler {
+    pub fn new() -> Self {
+        Self
+    }
+}
+impl miette::ReportHandler for ErrorReportHandler {
+    fn debug(&self, error: &dyn Diagnostic, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        if f.alternate() {
+            return core::fmt::Debug::fmt(error, f);
+        }
+        let code_as_str = match error.code() {
+            Some(code) => code.to_string(),
+            None => "OCK500".to_string(),
+        };
+
+        writeln!(
+            f,
+            "{} {}\n",
+            code_as_str
+                .color(crate::terminal::OckamColor::FmtERRORBackground.color())
+                .bold(),
+            error
+        )?;
+
+        if let Some(help) = error.help() {
+            writeln!(f, "{}", fmt_log!("{}", help))?;
+        }
+
+        if let Some(url) = error.url() {
+            writeln!(f, "{}", fmt_log!("{}", url))?;
+        }
```

Comment:
Until we have an actual doc setup to handle error codes, this can be hidden/removed.

Just keeping it for potential example/demo purposes at the moment.

Reply:
I've left that part as a TODO